src/Octoshift/Services/AzureApi.cs

```diff
@@ -89,4 +101,24 @@ private Uri GetServiceSasUriForBlob(BlobClient blobClient)
 
         return blobClient.GenerateSasUri(sasBuilder);
     }
+
+    private void LogProgress(long uploadedBytes, long totalBytes)
+    {
+        lock (_mutex)
+        {
+            if (DateTime.Now < _nextProgressReport)
+            {
+                return;
+            }
+
+            _nextProgressReport = _nextProgressReport.AddSeconds(UPLOAD_PROGRESS_REPORT_INTERVAL_IN_SECONDS);
+        }
+
+        var percentage = (int)(uploadedBytes * 100L / totalBytes);
```

Comment:
If archives are being transferred on a lossy connection, I believe that the percentage completed could exceed 100%.

Reply:
(Depending upon how/if TCP header size is removed from uploadedBytes, that could also cause a discrepancy)

Reply:
I don't think so as the `uploadedBytes` will always be reported by the Azure library. It should take care of everything including retries, etc internally.